src/java.base/windows/native/libnio/ch/UnixDomainSockets.c

```diff
@@ -84,21 +91,73 @@ jint unixSocketAddressToSockaddr(JNIEnv *env, jbyteArray addr, struct sockaddr_u
     return ret;
 }
 
+static int cmpGuid(GUID *g1, GUID *g2) {
+    if (g1->Data1 != g2->Data1)
+        return 0;
+    if (g1->Data2 != g2->Data2)
+        return 0;
+    if (g1->Data3 != g2->Data3)
+        return 0;
+    return !memcmp(g1->Data4, g2->Data4, 8);
```

Comment:
I think this would be easier to understand if Data4 were compared in the same way as Data1-Data3.

Reply:
I presume you mean compare each element of the array individually. I guess we can do that, as the function is only called once.